examples/vault-cluster-authentication-iam/startup-script-vault.sh

```diff
@@ -0,0 +1,145 @@
+function retry {
+  local -r cmd="$1"
+  local -r description="$2"
+
+  for i in $(seq 1 30); do
+    log "$description"
+
+    # The boolean operations with the exit status are there to temporarily circumvent the "set -e" at the
+    # beginning of this script which exits the script immediatelly for error status while not losing the exit status code
+    output=$(eval "$cmd") && exit_status=0 || exit_status=$?
+    log "$output"
+    if [[ $exit_status -eq 0 ]]; then
+      echo "$output"
+      return
+    fi
+    log "$description failed. Will sleep for 10 seconds and try again."
+    sleep 10
+  done;
+
+  log "$description failed after 30 attempts."
+  exit $exit_status
+}
+
+# Initializes a vault server
+# run-vault is running on the background and we have to wait for it to be done,
+# so in case this fails we retry.
+SERVER_OUTPUT=$(retry \
+  "/opt/vault/bin/vault operator init" \
+  "Trying to initialize vault")
+
+# The expected output should be similar to this:
+# ==========================================================================
+# Unseal Key 1: ddPRelXzh9BdgqIDqQO9K0ldtHIBmY9AqsTohM6zCRl7
+# Unseal Key 2: liSgypzdVrAxz73KbKyCMjVeSnRMuxCZMk1PWIZdjENS
+# Unseal Key 3: pmgeVu/fs8+jl8bOzf3Cq56BFufm4o7Sxt2oaUcvt6Dp
+# Unseal Key 4: i3W2xJEyUqUqcO1QSjTA+Ua0RUPxnNWM27AqaC8wW7Zh
+# Unseal Key 5: vHsQtCRgfblPeFYw1hhCVbji0MoNUP8zyIWhLWs3PebS
+#
+# Initial Root Token: cb076fc1-cc1f-6766-795f-b3822ba1ac57
+#
+# Vault initialized with 5 key shares and a key threshold of 3. Please securely
+# distribute the key shares printed above. When the Vault is re-sealed,
+# restarted, or stopped, you must supply at least 3 of these keys to unseal it
+# before it can start servicing requests.
+#
+# Vault does not store the generated master key. Without at least 3 key to
+# reconstruct the master key, Vault will remain permanently sealed!
+#
+# It is possible to generate new unseal keys, provided you have a quorum of
+# existing unseal keys shares. See "vault operator rekey" for more information.
+# ==========================================================================
+
+# Unseals the server with 3 keys from this output
+# Please note that this is not how it should be done in production as it is not secure and and we are
+# not storing any of the tokens, so in case it gets resealed, the tokens are lost and we wouldn't be able to unseal it again
+# Normally, an operator would SSH and unseal the server in each node manually or, ideally,
+# it should be auto unsealed https://www.vaultproject.io/docs/enterprise/auto-unseal/index.html
+# For this quick example specifically, we are just running one vault server and unsealing it like this
+# for simplicity as this example focuses on authentication and not on unsealing.
+# For a more detailed example on auto unsealing, check the vault enterprise example.
+echo "$SERVER_OUTPUT" | head -n 3 | awk '{ print $4; }' | xargs -l /opt/vault/bin/vault operator unseal
```

Comment:
Had to work to understand this sequence of pipes. Consider breaking into several steps with named variables to make it easier:

```bash
first_three_lines=$(echo "$SERVER_OUTPUT" | head -n 3)
unseal_keys=$(echo "$first_three_lines" | awk '{ print $4; }')
```